modules/system/utility-wfrp4e.js

```diff
@@ -1098,96 +1102,93 @@ export default class WFRP_Utility {
+          await WFRP_Utility.awaitSocket(game.user, "applyOneTimeEffect", payload, "invoking effect");
+          return
         }
       }
     }
 
-    WFRP_Utility.runSingleEffect(effect, actor, null, { actor }, {async : true});
+    await WFRP_Utility.runSingleEffect(effect, actor, null, { actor });
   }
 
-  static async runSingleEffect(effect, actor, item, scriptArgs, options = {}) {
-    try {
-      let func;
-      if (!options.async) {
-        func = new Function("args", effect.flags.wfrp4e.script).bind({ actor, effect, item })
-        WFRP_Utility.log(`${this.name} > Running ${effect.name}`)
-      } else if (options.async) {
-        let asyncFunction = Object.getPrototypeOf(async function () { }).constructor
-        func = new asyncFunction("args", effect.flags.wfrp4e.script).bind({ actor, effect, item })
-        WFRP_Utility.log(`${this.name} > Running Async ${effect.name}`)
+  static async runSingleEffect(effect, actor, item, scriptArgs) {
+      try {
+        if (WFRP_Utility.effectCanBeAsync(effect)) {
+          let asyncFunction = Object.getPrototypeOf(async function () { }).constructor
+          const func = new asyncFunction("args", effect.flags.wfrp4e.script).bind({ actor, effect, item })
+          WFRP_Utility.log(`${this.name} > Running Async ${effect.name}`)
+          await func(scriptArgs);
+        } else {
+          let func = new Function("args", effect.flags.wfrp4e.script).bind({ actor, effect, item })
+          WFRP_Utility.log(`${this.name} > Running ${effect.name}`)
+          func(scriptArgs);
+        }
       }
-      if (func) {
-        func(scriptArgs);
+      catch (ex) {
+        ui.notifications.error(game.i18n.format("ERROR.EFFECT", { effect: effect.name }))
+        console.error("Error when running effect " + effect.name + " - If this effect comes from an official module, try replacing the actor/item from the one in the compendium. If it still throws this error, please use the Bug Reporter and paste the details below, as well as selecting which module and 'Effect Report' as the label.")
+        console.error(`REPORT\n-------------------\nEFFECT:\t${effect.name}\nACTOR:\t${actor.name} - ${actor.id}\nERROR:\t${ex}`)
       }
-    }
-    catch (ex) {
-      ui.notifications.error(game.i18n.format("ERROR.EFFECT", { effect: effect.name }))
-      console.error("Error when running effect " + effect.name + " - If this effect comes from an official module, try replacing the actor/item from the one in the compendium. If it still throws this error, please use the Bug Reporter and paste the details below, as well as selecting which module and 'Effect Report' as the label.")
-      console.error(`REPORT\n-------------------\nEFFECT:\t${effect.name}\nACTOR:\t${actor.name} - ${actor.id}\nERROR:\t${ex}`)
-    }
   }
 
-  static invokeEffect(actor, effectId, itemId) {
+  static effectCanBeAsync (effect) {
+    return (game.wfrp4e.config.syncEffectTriggers.indexOf(effect.trigger) === -1)
```

Comment:
Could you not do `!game.wfrp4e.config.syncEffectTriggers.includes(effect.trigger)`? Seems clearer

Reply:
old javascript habits. i will change that.

Reply:
changed